hyper/common/headers.py

```diff
@@ -181,6 +181,19 @@ def iter_raw(self):
         for item in self._items:
             yield item
 
+    def replace(self, key, value):
+        """
+        Replace existing header with new value. If header doesn't exist this
+        method work like ``__setitem__``. Replacing leads to deletion of all 
+        exsiting headers with the same name.
+        """
+        try:
+            del self[key]
+        except KeyError:
+            pass
+
+        self._items.append(to_bytestring_tuple(key, value))
```

Comment:
At this point, I actually think we can make `replace` a simple synonym for `del` then `__setitem__`. That is, we can replace the inlined call to `__setitem__` (that is, `self._items.append`) with an actual call to `__setitem__` (`self[key] = value`), which should reduce our code duplication and reduce the risk of this breaking if the implementation changes later on.